src/Gtk/Avalonia.Cairo/Media/DrawingContext.cs

```diff
@@ -179,11 +179,25 @@ public void DrawGeometry(IBrush brush, Pen pen, Geometry geometry)
         /// <param name="rect">The rectangle bounds.</param>
         public void DrawRectangle(Pen pen, Rect rect, float cornerRadius)
         {
-            using (var p = SetPen(pen, rect.Size)) 
-            {
-                _context.Rectangle(rect.ToCairo ());
-                _context.Stroke();
-            }
+			if (cornerRadius == 0) {
+				using (var p = SetPen (pen, rect.Size)) {
```

Comment:
You can move the using statement to be outside the if\else block to reduce the duplication